hoomd/simulation.py

```diff
@@ -115,6 +116,30 @@ def state(self):
     def operations(self):
         return self._operations
 
+    @operations.setter
+    def operations(self, operations):
+        # This condition is necessary to allow for += and -= operators to work
+        # correctly with simulation.operations (+=/-=).
+        if operations is self._operations:
+            self._operations = operations
+        else:
+            # Handle error cases first
+            if operations.scheduled:
+                raise RuntimeError(
+                    "Cannot add scheduled `hoomd.Operations` object to "
+                    "`hoomd.Simulation` object.")
+            elif operations._simulation is not None:
+                raise RuntimeError(
+                    "Cannot add `hoomd.Operations` object that belongs to "
+                    "another `hoomd.Simulation` object.")
+            # Switch out `hoomd.Operations` objects.
+            elif self._operations.scheduled:
+                self._operations.unschedule()
+                self._operations._simulation = None
+                operations._simulation = self
+                operations.schedule()
+                self._operations = operations
+
     def sanity_check(self):
         raise NotImplementedError
```

Comment:
This seems like an oversight. This method should probably be private, or removed?

Reply:
It should be removed, but this is not really related to the PR at hand. There are multiple other methods that likely should be removed from `hoomd.simulation.Simulation`.

Reply:
#783 cleans removes these methods from `Simulation`